src/validator/RecoveryPlugin.sol

```diff
@@ -0,0 +1,202 @@
+        require(data.length % 65 == 0, "Data length must be a multiple of 65");
+
+        uint chunks = data.length / 65;
+
+        bytes[] memory dividedBytes = new bytes[](chunks);
+
+        for (uint i = 0; i < chunks; i++) {
+            bytes memory chunk = new bytes(65);
+            for (uint j = 0; j < 65; j++) {
+                chunk[j] = data[(i * 65) + j];
+            }
+            dividedBytes[i] = chunk;
+        }
+        return dividedBytes;
+    }
+
+    function changeOwner(address _newOwner) internal {
+        address oldOwner = recoveryPluginStorage[msg.sender].owner;
+        recoveryPluginStorage[msg.sender].owner = _newOwner;
+        for (uint256 i = 0; i < guardians[msg.sender].length; i++) {
+            guardians[msg.sender][i].approved = false;
+        }
+        emit OwnerChanged(msg.sender, oldOwner, _newOwner);
+    }
+
+    function addGuardian(
+        Guardian[] calldata _guardians,
+        uint256 _thresholdWeight,
+        uint256 delay
+    ) external {
+        for (uint256 i = 0; i < _guardians.length; i++) {
+            if (_guardians[i].weight <= 0) {
+                revert();
+            }
+            if (_guardians[i].guardian == address(0)) {
+                revert();
+            }
+            if (_guardians[i].guardian == msg.sender) {
+                revert();
+            }
+            if (_guardians[i].approved) {
+                revert();
+            }
+            guardians[msg.sender].push(_guardians[i]);
+        }
+        thresholdWeight[msg.sender] = _thresholdWeight;
+        recoveryDelay[msg.sender] = block.timestamp + delay;
+        emit GuardianAdded(msg.sender, _guardians, _thresholdWeight);
+    }
+
+    function verifyGuardians(
+        bytes32 hash,
+        bytes[] memory signatures
+    ) internal returns (uint256) {
+        uint256 weight = 0;
+        for (uint256 i = 0; i < signatures.length; i++) {
+            if (signatures[i].length != 65) {
+                revert();
+            } else {
+                if (
+                    validateGuardianSignature(
+                        hash,
+                        signatures[i],
+                        guardians[msg.sender][i].guardian
+                    ) ==
+                    0 &&
+                    !guardians[msg.sender][i].approved
+                ) {
+                    unchecked {
+                        weight += guardians[msg.sender][i].weight;
+                    }
+                }
+            }
+        }
+        return weight;
+    }
+
+    function initRecovery(
+        address _newOwner,
+        bytes32 hash,
+        bytes[] memory signatures
+    ) internal {
+        require(
+            _newOwner != address(0),
+            "RecoveryPlugin: new owner is zero address"
+        );
+        require(hash != bytes32(0), "RecoveryPlugin: hash is zero");
+        require(
+            block.timestamp >= recoveryDelay[msg.sender],
```

Comment:
accessing the timestamp will violate the 4337 rule so enable should not be called on validation phase right? which also makes it impossible for wallet to set this as default on deployment

Reply:
maybe we can do short circuit here like `require(oldOwner == address(0) || ~)`